### ISSUE TITLE
Support returning extension outputs

### ISSUE THREAD

**Issue by MasterKale** [MasterKale/SimpleWebAuthn#22]
While preparing #21, I discovered that **browser** wasn't actually returning extension results due to some weird issue with `navigator.credentials.create()`/`get()` returning a `credential` with non-enumerable properties, so the spread operator was ineffective.

Anyway after some more digging it turns out that TypeScript's [AuthenticationExtensionsClientInputs](https://github.com/microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L156) and [AuthenticationExtensionsClientOutputs](https://github.com/microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L167) definitions are wildly out of date, especially due to a recent WebAuthn spec change [that removed a bunch of outdated extensions](https://github.com/w3c/webauthn/pull/1399). Therefore I need to upgrade **browser** to support returning the results of `getClientExtensionResults()` to support the list of extension [currently defined in the spec](https://w3c.github.io/webauthn/#sctn-defined-extensions):

- appid
- appidExclude
- uvm
- credProps

**Comment by lgarron** [comment]
FYI, we have updated types at:

https://github.com/github/webauthn-json/blob/main/src/extended/json.ts
https://github.com/github/webauthn-json/blob/main/src/extended/schema.ts

TypeScript's types tend to lag significantly behind the spec (and Flow even more so), so we probably need to maintain our own list for the near future.

**Comment by lgarron** [comment]
Also, note that we might be getting a new extension soon:

**Comment by MasterKale** [comment]
@lgarron You rock, thank you for pointing that out!

Man, I should just replace **@simplewebauthn/browser** with **webauthn-json**, you've all already taken care of this 😂

**Comment by MasterKale** [comment]
That said I see an opportunity to combine **@simplewebauthn/typescript-types** and **webauthn-json** and compose a single, definitive types-only library that can remain up-to-date as the spec evolves. Maybe it becomes a repo in the WACG for communal maintenance?

**Comment by mchorvat** [comment]
Matt, are txAuthSimple and txAuthGeneric extensions working fine in the latest build? Thanks for all your great work!

**Comment by MasterKale** [comment]
Extension support is only partially functional. `generateAttestationOptions()` and `generateAssertionOptions()` accept an `extensions` argument that gets passed straight through to the returned object without manipulation. There shouldn't be any issues with passing extensions to authenticators.

Where things fall down is getting the resulting extension output _back_ to the RP so that they can bed used. @simplewebauthn/browser still needs to be updated to return the return value from `getClientExtensionResults()`.

**Comment by Moumouls** [comment]
Hi 🙂 

Do we have some updates here about extensions handling (uvm and others) ?

Is there a problem extracting "extensions" here ?

```ts
  const { id, rawId, response, type } = credential;

  let userHandle = undefined;
  if (response.userHandle) {
    userHandle = bufferToBase64URLString(response.userHandle);
  }

  // Convert values to base64 to make it easier to send back to the server
  return {
    id,
    rawId: bufferToBase64URLString(rawId),
    response: {
      authenticatorData: bufferToBase64URLString(response.authenticatorData),
      clientDataJSON: bufferToBase64URLString(response.clientDataJSON),
      signature: bufferToBase64URLString(response.signature),
      userHandle,
    },
    type,
  };
```

**Comment by MasterKale** [comment]
As a matter of fact I started work on this before real-world stuff came up. Check out the previously-stale [feature/browser-extensions-support](https://github.com/MasterKale/SimpleWebAuthn/tree/feature/browser-extensions-support) branch (that I just rebased on the latest master this morning), it includes the start of the effort to handle extension input and output.

To make it easier to follow progress I've created a draft PR #94 for that branch.

As for an ETA, this has been on the back of my mind the whole time and so I _want_ to get it done soon but have no idea when it'll actually land 😅

**Comment by MasterKale** [comment]
@Moumouls PR #94 just landed and has been published as v2.1.0. **@simplewebauthn/browser@2.1.0** will now return back the value of `credential.getClientExtensionResults()` as the new `clientExtensionResults` property on the values that are returned by both `startAttestation()` and `startAssertion()`.
